### PR TITLE
feat: add filecoin chain

### LINKED ISSUE #673: filecoin integration into stable

### PR DESCRIPTION
**Pull request by davidmurdoch**
Resolved #673

### CODE REVIEW

File: src/chains/filecoin/src/api.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+  constructor(blockchain: Blockchain) {
+    this[_blockchain] = blockchain;
+  }
+
+  async stop(): Promise<void> {
+    return await this[_blockchain].stop();
+  }
+
+  async "Filecoin.ChainGetGenesis"(): Promise<SerializedTipset> {
```
Comment (seesemichaelj) [inline src/chains/filecoin/src/api.ts:33]: @davidmurdoch @tcoulter why do these function names need periods, requiring them to be declared with string literals? Why not just have `FilecoinChainGetGenesis` or `Filecoin_ChainGetGenesis`?

**Comment by tcoulter** [comment]
That’s how the Filecoin/Lotus API works. The RPC method names are literally
`Filecoin.X`, where X is the individual method you’re calling.

File: src/chains/filecoin/src/api.ts (hunk 1 above)
Comment (seesemichaelj) [reply src/chains/filecoin/src/api.ts:33]: Thanks @tcoulter for the [clarification](https://github.com/trufflesuite/ganache-core/pull/681#issuecomment-749111969)!

**Comment by seesemichaelj** [comment]
I'm closing this PR in favor of one that I will open

**Pull request closed without merging**